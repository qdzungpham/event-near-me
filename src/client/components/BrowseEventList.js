import React from 'react';
import { Link } from 'react-router-dom';

import BrowseEventCard from './BrowseEventCard';

const BrowseEventList = ({ events, overMarker }) => events.map(event => (
  <Link key={event.id} to={`/event/${event.id}`} style={{ textDecoration: 'none' }}>
    <BrowseEventCard event={event} overMarker={overMarker} />
  </Link>
));

export default BrowseEventList;
