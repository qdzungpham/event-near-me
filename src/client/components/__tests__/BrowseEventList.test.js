import React from 'react';
import { shallow } from 'enzyme';
import BrowseEventList from '../BrowseEventList';

describe('BrowseEventList', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <BrowseEventList
        title="test"
        events={[
          {
            id: '2',
            title: 'title',
            dateTime: '2018-09-07T14:12:15.112Z',
            location: 'Brisbane',
            coords: [-24, 153],
            distance: 21,
            image: 'http://via.placeholder.com/100x100'
          }
        ]}
        overMarker="-1"
        onHoverEventCard={jest.fn}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
