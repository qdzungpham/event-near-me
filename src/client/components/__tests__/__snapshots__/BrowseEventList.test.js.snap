// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrowseEventList renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BrowseEventList
    events={
      Array [
        Object {
          "coords": Array [
            -24,
            153,
          ],
          "dateTime": "2018-09-07T14:12:15.112Z",
          "distance": 21,
          "id": "2",
          "image": "http://via.placeholder.com/100x100",
          "location": "Brisbane",
          "title": "title",
        },
      ]
    }
    onHoverEventCard={[Function]}
    overMarker="-1"
    title="test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "2",
    "nodeType": "class",
    "props": Object {
      "children": <WithStyles(BrowseEventCard)
        event={
          Object {
            "coords": Array [
              -24,
              153,
            ],
            "dateTime": "2018-09-07T14:12:15.112Z",
            "distance": 21,
            "id": "2",
            "image": "http://via.placeholder.com/100x100",
            "location": "Brisbane",
            "title": "title",
          }
        }
        onHoverEventCard={[Function]}
        overMarker="-1"
      />,
      "replace": false,
      "style": Object {
        "textDecoration": "none",
      },
      "to": "/event/2",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "event": Object {
          "coords": Array [
            -24,
            153,
          ],
          "dateTime": "2018-09-07T14:12:15.112Z",
          "distance": 21,
          "id": "2",
          "image": "http://via.placeholder.com/100x100",
          "location": "Brisbane",
          "title": "title",
        },
        "onHoverEventCard": [Function],
        "overMarker": "-1",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "2",
      "nodeType": "class",
      "props": Object {
        "children": <WithStyles(BrowseEventCard)
          event={
            Object {
              "coords": Array [
                -24,
                153,
              ],
              "dateTime": "2018-09-07T14:12:15.112Z",
              "distance": 21,
              "id": "2",
              "image": "http://via.placeholder.com/100x100",
              "location": "Brisbane",
              "title": "title",
            }
          }
          onHoverEventCard={[Function]}
          overMarker="-1"
        />,
        "replace": false,
        "style": Object {
          "textDecoration": "none",
        },
        "to": "/event/2",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "event": Object {
            "coords": Array [
              -24,
              153,
            ],
            "dateTime": "2018-09-07T14:12:15.112Z",
            "distance": 21,
            "id": "2",
            "image": "http://via.placeholder.com/100x100",
            "location": "Brisbane",
            "title": "title",
          },
          "onHoverEventCard": [Function],
          "overMarker": "-1",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
